src/Presentation/Controllers/UserController.cs

```diff
@@ -0,0 +1,96 @@
+using Microsoft.AspNetCore.Mvc;
+using ExpenseTrackerGrupo4.src.Domain.Entities;
+using ExpenseTrackerGrupo4.src.Aplication.Interfaces;
+using Microsoft.AspNetCore.Authorization;
+using ExpenseTrackerGrupo4.src.Utils;
+using ExpenseTrackerGrupo4.src.Presentation.DTOs;
+using AutoMapper;
+
+namespace ExpenseTrackerGrupo4.src.Presentation.Controllers;
+
+[Route("api/[controller]")]
+[ApiController]
+[Authorize]
+public class UserController(IUserService userService, IMapper mapper) : ControllerBase
+{
+    private readonly IUserService _userService = userService;
+    private readonly IMapper _mapper = mapper;
+
+    [HttpGet("{id}")]
+    public async Task<IActionResult> GetUserById(Guid id)
+    {
+        var user = await _userService.GetUserByIdAsync(id);
+
+        if(user == null)
+        {
+            return NotFound(new { Message = "User not found." });
+        }
+
+        return Ok(user);
+    }
+
+    [HttpPut]
+    public async Task<IActionResult> UpdateUser([FromBody] UserUpdateRequestDTO userUpdated)
```

Comment:
The id should not be requested in the body, it should be a field like getById or Delete